src/right-field/RightField.jsx

```diff
@@ -1,26 +1,183 @@
- * @returns {Component} A React component that displays the given query in a table, with the functionality to refresh the results and to login for additional authorization.  
+ *
+ * @param {query} props.query The query (as defined in the config file) that should be executed and results displayed in the table.
+ * @returns {Component} A React component that displays the given query in a table, with the functionality to refresh the results and to login for additional authorization.
  */
-function RightField(props){
-    let query = props.query
-    let refreshRef = useRef(undefined) 
-    return(
-        <div className="right-field" style={{backgroundColor: config.mainAppColor}}>
-            <div className="control-section">
-                <button ref={refreshRef}  disabled={query===undefined} id="refresh-button">Refresh</button>
-                {query && <label id="query-name-label">{query.name}</label>}
-                <button id="login-button">Login</button>
-            </div>
-            <ResultsTable refreshButton={refreshRef}  selectedQuery={query}/>
+function RightField(props) {
+  let selectedQuery = props.query;
+  const [results, setResults] = useState([]);
+  const [variables, setVariables] = useState([]);
+  const [isQuerying, setQuerying] = useState(false);
+  const [time, setTime] = useState(0);
+
+  useEffect(() => {
+    let intervalId;
+    if (isQuerying) {
+      intervalId = setInterval(() => setTime(time + 1), 10);
+    }
+    return () => clearInterval(intervalId);
+  }, [time, isQuerying]);
+
+  useEffect(() => {
+    configureQueryWorker(adder, setVariables, setQuerying);
+  }, []);
+
+  let adder = adderFunctionMapper["bindings"](setResults);
+
+  const onQueryChanged = () => {
+    setTime(0);
+    if (selectedQuery) {
+      if (isQuerying) {
+        queryWorker.terminate();
+        configureQueryWorker(adder, setVariables, setQuerying);
+      }
+      setResults([]);
+      setVariables([]);
+      setQuerying(true);
+      executeQuery(selectedQuery, queryWorker);
+    }
+  };
+
+  useEffect(() => {
+    onQueryChanged();
+  }, [selectedQuery]);
+
+  return (
+    <div
+      className="right-field"
+      style={{ backgroundColor: config.mainAppColor }}
+    >
+      <div className="control-section">
+        <button
+          disabled={!selectedQuery}
+          id="refresh-button"
+          onClick={onQueryChanged}
+        >
+          Refresh
+        </button>
+        <div id="query-information">
+          {selectedQuery && (
+            <label>
+              <strong>Result Count:</strong>
+              {results.length}
+            </label>
+          )}
+          {selectedQuery && (
+            <strong id="query-name-label">{selectedQuery.name}</strong>
+          )}
+          {selectedQuery && (
+            <label className="stopWatch">
+              <strong>Runtime:</strong>
+              <Time time={time} showMilliseconds={config.showMilliseconds} />
```

Comment:
Once the query is actually done, it should also explicitly mention that.

Reply:
I have done this by replacing "Runtime: " with "Finished in:" in [commit 7e15f24](https://github.com/SolidLabResearch/generic-data-viewer/pull/17/commits/7e15f24a8eb9cbc311fb3d8b90e26aa07c142398). Is this change obvious and user friendly?

Reply:
That's good!